proxy_agent/src/proxy/proxy_connection.rs

```diff
@@ -3,23 +3,40 @@
 
 //! This module contains the connection context struct for the proxy listener, and write proxy processing logs to local file.
 
+use crate::common::hyper_client;
 use crate::proxy::Claims;
 use proxy_agent_shared::{logger_manager, rolling_logger::RollingLogger};
-use std::net::TcpStream;
+use std::net::{Ipv4Addr, SocketAddr, TcpStream};
 use std::path::PathBuf;
 use std::sync::{Arc, Mutex};
 use std::time::Instant;
 
-pub struct Connection {
-    pub stream: TcpStream,
+pub struct ConnectionContext {
     pub id: u128,
-
+    pub stream: Arc<Mutex<TcpStream>>,
+    pub client_addr: SocketAddr,
     pub now: Instant,
-    pub cliams: Option<Claims>,
-    pub ip: String,
+    pub claims: Option<Claims>,
+    pub method: hyper::Method,
+    pub url: hyper::Uri,
+    pub ip: Option<Ipv4Addr>, // currently, we only support IPv4
     pub port: u16,
 }
 
+impl ConnectionContext {
+    pub fn should_skip_sig(&self) -> bool {
+        hyper_client::should_skip_sig(self.method.clone(), self.url.clone())
+    }
+
+    pub fn request_ip(&self) -> String {
```

Comment:
Why string? `Option` would generally be better and should work if this is related to serialization or logging

Reply:
this is for logging purposes. let me update the fn name and add bit more comments to it too.